File: C#/Programming Basics with C#/11. For-Loop - Exercise/02. Half Sum Element/HalfSumElement.cs
using System;

class HalfSumElement
{
    static void Main()
    {
        int n = int.Parse(Console.ReadLine());
        int num = 0;
        int sum = 0;
        int maxNum = 0;
        for (int i = 0; i < n; i++)
        {
            num = int.Parse(Console.ReadLine());
            sum = sum + num;
            if (maxNum < num)
            {
                maxNum = num;
            }
        }
        if (maxNum == (sum - maxNum))
        {
            Console.WriteLine("Yes");
            Console.WriteLine($"Sum = {sum - maxNum}");
        }
        else
        {
            Console.WriteLine("No");
            Console.WriteLine($"Diff = {Math.Abs(sum - maxNum * 2)}");
        }
    }
}

File: C++/Programming Basics with C++/11. Exercise - For-loop/simple_tasks/HalfSumElement.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
    int n;
    cin >> n;

    int sum = 0;
    int maxNum = 0;

    for (int i = 0; i < n; i++) {
        int num;
        cin >> num;
        sum += num;

        if (maxNum < num) {
            maxNum = num;
        }
    }
    if (sum - maxNum == maxNum) {
        cout << "Yes" << endl;
        cout << "Sum = " << maxNum << endl;
    } else {
        cout << "No" << endl;
        cout << "Diff = " << abs(sum - 2 * maxNum) << endl;
    }

    return 0;
}
